wdl-lint/src/rules/disallowed_declaration_name.rs

```diff
@@ -0,0 +1,345 @@
+                }
+                PrimitiveTypeKind::Float => {
+                    type_names.insert(primitive_type.to_string());
+                }
+                PrimitiveTypeKind::Directory => {
+                    type_names.insert(primitive_type.to_string());
+                    // Also check for "Dir"
+                    type_names.insert("Dir".to_string());
+                }
+            }
+        }
+        Type::Array(_) => {
+            // Add "Array" for the compound type
+            type_names.insert("Array".to_string());
+        }
+        Type::Map(_) => {
+            // Add "Map" for the compound type
+            type_names.insert("Map".to_string());
+        }
+        Type::Pair(_) => {
+            // Add "Pair" for the compound type
+            type_names.insert("Pair".to_string());
+        }
+        Type::Object(_) => {
+            // Add "Object" for the object type
+            type_names.insert("Object".to_string());
+        }
+    }
+
+    // Get declaration name after type checking (moving closer to usage)
+    let name = decl.name();
+    let name_str = name.as_str();
+
+    // Check if the declaration name ends with one of the type names
+    for type_name in &type_names {
+        let type_lower = type_name.to_lowercase();
+
+        // Special handling for short type names (3 characters or less)
+        // These require word-based checks to avoid false positives
+        if type_lower.len() <= 3 {
+            // Split the identifier into words
+            let words = split_to_words(name_str);
+
+            // Check if the short type name appears as the last word
+            if let Some(last_word) = words.last() {
+                if last_word.to_lowercase() == type_lower {
+                    state.exceptable_add(
+                        decl_identifier_with_type(decl.name().span(), name_str, type_name),
+                        SyntaxElement::from(decl.syntax().clone()),
+                        exceptable_nodes,
+                    );
+                    return;
+                }
+            }
+        } else {
+            // For longer types, check if the identifier ends with the type name
+            let name_lower = name_str.to_lowercase();
+            if name_lower.ends_with(&type_lower) {
+                state.exceptable_add(
+                    decl_identifier_with_type(decl.name().span(), name_str, type_name),
+                    SyntaxElement::from(decl.syntax().clone()),
+                    exceptable_nodes,
+                );
+                return;
+            }
+        }
+    }
+}
+
+/// Split an identifier into words using convert_case
+fn split_to_words(identifier: &str) -> Vec<String> {
+    // Use convert_case's built-in split functionality with default boundaries
+    convert_case::split(&identifier, &convert_case::Boundary::defaults())
+        .into_iter()
+        .map(|s| s.to_string())
+        .collect()
+}
+
+#[cfg(test)]
```

Comment:
The tests you've written here are essentially the same as the tests checked by `source.wdl` and `source.errors`, so I think it's ok to just delete all these.